Dockerfile.railgun

```diff
@@ -2,20 +2,12 @@
 FROM golang:1.16 as builder
```

Comment:
I'm not sure I understand why we're moving this root level: thus far we've made railgun available in KIC, but isolated to its directory until we're ready to release. What was the impetus for moving the Dockerfile at this time?

Reply:
The reason for moving the Dockerfile was:
1. The docker build context needs to include the repo root (meaning `docker build -f $dockerfile $repo_root` as opposed to `docker build -f $dockerfile $repo_root/railgun`) level because the go module is defined there, and railgun makes use of packages defined outside of `railgun/`.
2. My preference for all Dockerfiles in a project to be defined in a single location (either the root directory of the build context, or `/build`)

At the same time I understand your way of thinking about isolating Railgun-specific stuff in the `railgun` directory, but I saw it as more of legacy (because railgun lived in a separate repo back in its early days) than a conscious decision - I was probably mistaken here. On our way to 2.0 we've been making strides to blend the 1.x codebase with railgun:
- reusing the parser/store logic
- making the integration test universal
- merging the Go modules
- (not yet done) unifying the build process

I think that we need to find agreement on whether this isolation is something we want to keep, and for how long. I can see two possible approaches:
- maintain railgun as autonomous and separate from KIC's codebase (and SDLC) as possible, including build system, dockerfiles, etc., and make a hard cut at 2.0, replacing the old with the new
- integrate railgun's and KIC 1.x's codebases and gradually retire 1.x stuff. Release 2.0 as if we were releasing 1.x, only substituting one compiled executable for another.

You seem to be coming from the former, I'm coming from the latter paradigm.

Reply:
The version you reviewed mistakenly removed the feature of caching downloaded modules idependently of code. d303594 reintroduces that.

Reply:
:+1: